pallene/parser.lua

```diff
@@ -17,6 +17,8 @@ function Parser:init(lexer)
     self.loop_depth = 0
     self.next = false -- Token
     self.look = false -- Token
+    self.regions = {} -- Sequence of sequence
+    self.region_depth = 0
```

Comment:
I think it is better if the `self.next` and `self.look` are next to the calls to `self:_advance()`. Maybe we should reorder  the variables to be the same order as the methods.

```
self.next = false -- Token
self.look = false -- Token
self:_advance(); self:_advance()
self.loop_depth = 0
self.region_depth = 0
self.regions = {}
```